packages/components/src/Collapse.tsx

```diff
@@ -4,52 +4,74 @@ import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import { CSSTransition } from 'react-transition-group';
 
-class Collapse extends Component {
-  static handleEnter(elemParam) {
+interface CollapseProps {
+  className?: string;
+  in: boolean;
+  children: React.ReactNode;
+  autoFocusOnShow?: boolean;
+}
+
+class Collapse extends Component<CollapseProps> {
+  static propTypes = {
+    className: PropTypes.string,
+    in: PropTypes.bool.isRequired,
+    children: PropTypes.node.isRequired,
+    autoFocusOnShow: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    className: '',
+    autoFocusOnShow: false,
+  };
+
+  static handleEnter(elemParam: HTMLElement): void {
     const elem = elemParam;
-    elem.style.height = 0;
+    elem.style.height = '0';
   }
 
-  static handleEntering(elemParam) {
+  static handleEntering(elemParam: HTMLElement): void {
     const elem = elemParam;
     elem.style.height = `${Collapse.getHeight(elem)}px`;
   }
 
-  static handleExiting(elemParam) {
+  static handleExiting(elemParam: HTMLElement): void {
     const elem = elemParam;
```

Comment:
I question the need for these consts - I realize this is a conversion so feel free to ignore.

Reply:
The const is an eslint thing I think. Pretty sure it would complain about changing param properties (but this is ok apparently)